From f95d4225610ac8a6f5b7ad270ef5f3b7853b78c4 Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Mon, 19 Nov 2018 09:58:32 +0200
Subject: [PATCH 0034/1239] phy: marvell cp110: mark u-boot power-off calls

It helps ATF to determine who called power off
function (U-boot/Linux) and act accordingly

Change-Id: Icfc5cbfdba64754496812154272b28c0ff639f0f
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-by: Grzegorz Jaszczyk <jaz@semihalf.com>
---
 drivers/phy/marvell/comphy_cp110.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/phy/marvell/comphy_cp110.c b/drivers/phy/marvell/comphy_cp110.c
index 39d173087b..63cba8a39e 100644
--- a/drivers/phy/marvell/comphy_cp110.c
+++ b/drivers/phy/marvell/comphy_cp110.c
@@ -549,10 +549,10 @@ int comphy_cp110_init(struct chip_serdes_phy_config *ptr_chip_cfg,
 		}
 		switch (ptr_comphy_map->type) {
 		case COMPHY_TYPE_UNCONNECTED:
+			mode = COMPHY_TYPE_UNCONNECTED | COMPHY_CALLER_UBOOT;
 			ret = comphy_smc(MV_SIP_COMPHY_POWER_OFF,
 					 ptr_chip_cfg->comphy_base_addr,
-					 lane,
-					 ptr_comphy_map->type);
+					 lane, mode);
 		case COMPHY_TYPE_IGNORE:
 			continue;
 			break;
-- 
2.29.0

